### PR TITLE
Feat: delete draft transfer #228

### PR DESCRIPTION
**Pull request by kevin-hashimoto**
#228 

- refactor: overhaul of the `AddEditTransfer` component and its child components
- feat: confirmation modal
- feat: button cluster configuration and display
- feat: new endpoints

TODO:
- ~~Transaction page grid display ( #409 related? )~~
  - ~~Create the endpoint for paginated transactions~~
  - ~~Alter the grid component on the Transactions page to display the paginated transaction data~~
- ~~Transaction page alert display~~
  - ~~detect alerts when a user lands on the Transactions page.~~
  
  
  Notes:
- The new patterns in the `AddEditTransfer` component should set us up to easily integrate the rest of the transfers features. This should be merged in before starting other components of the transfers.
- Different states of the child component UIs should be determined by checking the status of the transfer. `transferData?.transfer_status.status`

Usage:
- `buttonClusterConfig` determines the display and functionality of the buttons that should be displayed at the bottom of the transfer pages.
  - each key of the config object represents the state that the transfer is in.
  - check `buttonConfigs.js` to see examples of how the ui configuration is set for the buttons
  - attach a `handler` to the button that calls a function to do something
    - You'll have access to the form data in the handler. 
    - For buttons that open a modal, set the modal data with `setModalData`. It takes `onConfirm`, `buttonText`, `text` properties.
  - You can set the button as `disabled` as well
- There are various mutation functions:
  - `createDraft`: creates a new draft transfer.
  - `updateDraft`: updates an already existing draft transfer but does not update its status.
  - `updateTransfer`: updates an already existing transfer, updates status and comments if there are any
- Depending on what happens after each step, the `updateTransfer` may have to be slightly modify to accommodate.
- Within each of the child form components ( eg. `TransferDetails`, `AgreementDate` etc. ), we can determine Transfer status with  the `useTransfer` hook, then modify ui based on that ( changing input field to normal text etc ).

### CODE REVIEW

File: backend/lcfs/web/api/transfer/services.py [hunk 1]
```diff
@@ -106,4 +112,20 @@ async def update_transfer(self, transfer_data: TransferUpdate) -> TransferSchema
...
\ No newline at end of file
+        return TransferSchema.model_validate(updated_transfer)
+
+    @service_handler
+    async def update_transfer(self, transfer_id: int, transfer_data: TransferUpdate) -> TransferSchema:
+        transfer = await self.repo.get_transfer_by_id(transfer_id)
+        if not transfer:
+            raise DataNotFoundException(
+                f"Transfer with ID {transfer_id} not found")
+        transfer.transfer_status_id = transfer_data.transfer_status_id
```
Comment (AlexZorkin) [inline backend/lcfs/web/api/transfer/services.py:123]: this value has recently been updated to current_status_id so we'll need to swap that out here. Reason being is the TransferHistory table will keep the change history for each transfer, ie when their status changes. Please add the creation of a new TransferHistory record here with the new status, and set current_status on the Transfer record.

File: backend/lcfs/web/api/transfer/services.py
Comment (AlexZorkin) [inline backend/lcfs/web/api/transfer/services.py:1]: Please make sure the backend unit tests are working with these changes, and add any additional coverage needed for the update_transfer method.

File: backend/lcfs/web/api/transfer/views.py
Comment (AlexZorkin) [inline backend/lcfs/web/api/transfer/views.py:1]: add/update the backend unit tests for update_transfer_draft

**Review by AlexZorkin — CHANGES_REQUESTED** [review]
Looks great. You'll have to merge release into this branch first because there are a few breaking changes that need to be fixed around transfer_status getting renamed to current_status. Mentioned in comments.

**Comment by kevin-hashimoto** [comment]
@AlexZorkin fixes addressed

**Pull request merged**